shell/platform/android/io/flutter/plugin/common/MethodCall.java

```diff
@@ -0,0 +1,29 @@
+package io.flutter.plugin.common;
+
+import java.util.Objects;
+
+/**
+ * Command object representing a method call on a {@link FlutterChannel}.
+ */
+public final class MethodCall {
+    /**
+     * The name of the called method.
+     */
+    public final String method;
+
+    /**
+     * Arguments for the call.
+     */
+    public final Object arguments;
```

Comment:
Not sure I follow you here.

Do we have a good reason for enforcing arguments to be held in an array (or list or map) across all methods invoked by the JSON codec?

I'd like us to allow any JSON argument whatsoever: that could be null, a bool, a number, a string, a list, or a map, decided on a method-per-method basis. As in Dart, sometimes multiple arguments are best handled positionally with a list, sometimes as named with a map. And zero/single argument functions should ideally not have to pay for a wrapper they don't use.

Implemented using JSONObject and friends, that means the static type is Object. (With the JEE7 JsonValue type hierarchy everything is wrapped in a JsonValue sub-type, but I don't think that library is present on Android(?), and it does imply an extra object wrapping around the meat).

Separation of concerns if often helpful, but splitting codecs for messages and method calls (and event streams?) also has a price. Do we then also want multiple channel concepts (or does it even make sense then), or do we configure each channel with two or three different codecs?